src/rgw/rgw_rest_pubsub.cc

```diff
@@ -463,6 +743,30 @@ void RGWPSDeleteTopicOp::execute(optional_yield y) {
     return;
   }
   const RGWPubSub ps(driver, s->owner.get_id().tenant);
+  rgw_pubsub_topic result;
+  op_ret = ps.get_topic(this, topic_name, result, y);
+  if (op_ret == 0) {
+    op_ret = verify_topic_owner_or_policy(
+        s, result, driver->get_zone()->get_zonegroup().get_name(),
+        rgw::IAM::snsDeleteTopic);
+    if (op_ret != 0) {
+      ldpp_dout(this, 1) << "failed to remove topic '" << topic_name
+                         << "' topic owned by other user" << dendl;
+      return;
+    }
+  } else {
+    ldpp_dout(this, 1) << "failed to fetch topic '" << topic_name << dendl;
```

Comment:
please change `op_ret` to `0`
currently the `test_ps_s3_notification_errors_on_master` test is failing since it gets `http status=404`.
the expected behavior is that if we delete an already deleted topic or a non-existent topic we should get `http statu=200`

Reply:
how about changing the test then, it makes sense to return 404 for a topic that did not exist ??

Reply:
the existing clients are not expceting a 404 on deleting an unknown entity (topic, object, bucket etc.)
we should keep the existing behavior, and return success in this case

Reply:
i just double checked to delete a non-existent bucket and i get 404
` s3cmd ls --verbose
2023-10-19 14:25  s3://testbkt

>> s3cmd rb s3://test --verbose
ERROR: S3 error: 404 (NoSuchBucket)
`

Reply:
> i just double checked to delete a non-existent bucket and i get 404 ` s3cmd ls --verbose 2023-10-19 14:25 s3://testbkt
> 
> > > s3cmd rb s3://test --verbose
> > > ERROR: S3 error: 404 (NoSuchBucket)
> > > `

as can be seen here in the code --> https://github.com/ceph/ceph/blob/main/src/rgw/rgw_op.cc#L3527-L3530

Reply:
you are right regarding bucket deletion, but there are other places where we consider such deletions as success.
look for code that looks like that
```
if (op_ret == -ENOENT) {
  op_ret = 0;
}
```
regardles, the existing behavior regarding topic deletion is to consider such deletion as success.

Reply:
> you are right regarding bucket deletion, but there are other places where we consider such deletions as success. look for code that looks like that
> 
> ```
> if (op_ret == -ENOENT) {
>   op_ret = 0;
> }
> ```
> 
> regardles, the existing behavior regarding topic deletion is to consider such deletion as success.

done